traitsui/testing/tester/ui_tester.py

```diff
@@ -0,0 +1,260 @@
+    effects are clicking or double clicking a mouse button, typing some keys
+    on the keyboard, etc.
+
+    To perform an interaction for side effects, call the ``UIWrapper.perform``
+    method with the interaction required. A set of predefined command types can
+    be found in the ``traitsui.testing.tester.command``.
+
+    Example::
+
+        with tester.create_ui(app, dict(view=view)) as ui:
+            tester.find_by_name(ui, "button").perform(command.MouseClick())
+            assert app.clicked
+
+    Inspecting GUI element (queries)
+    --------------------------------
+    Sometimes, the test may want to inspect visual elements on the GUI, e.g.
+    checking if a textbox has displayed the expected value.
+
+    To perform an interaction for queries, call the ``UIWrapper.inspect``
+    method with the query required. A set of predefined query types can be
+    found in the ``traitsui.testing.tester.query``.
+
+    Example::
+
+        with tester.create_ui(app, dict(view=view)) as ui:
+            text = (
+                tester.find_by_name(ui, "text").inspect(query.DisplayedText())
+            )
+            assert text == "Hello"
+
+    Extending the API
+    -----------------
+    THe API can be extended by defining a registry for mapping target and
+    interaction types to a specific implementation that handles the
+    interaction.
+
+    For example, suppose there is a custom UI target ``MyEditor``, to implement
+    a custom interaction type ``ManyMouseClick`` for this target::
+
+        custom_registry = InteractionRegistry()
+        custom_registry.register_handler(
+            target_class=MyEditor,
+            interaction_class=ManyMouseClick,
+            handler=lambda wrapper, interaction: wrapper.target.do_something()
+        )
+
+    Then the registry can be used in a UITester::
+
+        tester = UITester(interaction_registries=[custom_registry])
+
+    This is how TraitsUI supplies testing support for specific editors; the
+    default setup of ``UITester`` comes with a registry for testing TraitsUI
+    editors.
+
+    Similar the location resolution logic can be extended via the
+    ``location_registries`` parameter.
+
+    See documentation on ``InteractionRegistry`` and ``LocationRegistry`` for
+    details.
+
+    Attributes
+    ----------
+    interaction_registries : list of InteractionRegistry, optional
+        Registries of interaction for different target, in the order
+        of decreasing priority. A shallow copy will be made.
+    location_registries : list of LocationRegistry, optional
+        Registries for resolving nested UI targets, in the order
+        of decreasing priority. A shallow copy will be made.
+        Note that an additional registry for resolving editors in a
+        traitsui UI object is always added.
+    """
+
+    def __init__(self, interaction_registries=None, location_registries=None):
```

Comment:
> The original EditorActionRegistry is now renamed and split into two separate registries, one responsible for interactions (perform, inspect), another one responsible for locating nested target (locate). I am actually in minds whether to combine them into just one registry again..

I think I am going back on this: I will combine the registries and let composition occur there. It will make this argument list more manageable. In so far, one does need to modify both registries at the same time while adding testing supporting to a particular target.

I will push this change, but that would be it (and will still be ready for review).

Reply:
Done. Now I have to name the registry differently. Still open to the suggestion of separating them again.